File: 2015/01_Not_Quite_Lisp/part_2/main.js
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 1.2
 */

let input = null;

try {
  input = require('fs').readFileSync("../input.txt", "utf-8").split("");
} catch (err) {
  console.error(err);
}

if (input !== null) {
  let floor = 0; // Start at floor 0

  for (let i = 0; i < input.length; ++i) {
    /*
      '(' = ascend floor
      ')' = descend floor
     */
    floor = input.shift() == '(' ? floor + 1 : floor - 1;

    // If we hit the basement (-1), we found the position
    if (floor < 0) {
      console.log(i+1); // Position starts at 1
      break;
    }
  }
}
else console.error("Failed to read input.txt");

File: 2015/01_Not_Quite_Lisp/part_2/main.f90
!!!
! Author: happ_2h https://github.com/happ-2h
!
! Solution for Advent of Code Day 1.2
!!!

program notquitelisp
  implicit none

  integer :: floor, i, opstat
  character(len=7000) :: cc

  floor = 0
  i = 1

  open(1, file="../input.txt", status="old", action="read", iostat=opstat)
  if (opstat > 0) stop "Cannot open file input.txt"

    read(1, *) cc

    ! While above basement level
    do while (floor .ge. 0)
      if (cc(i:i) .eq. "(") then ! Ascend
        floor = floor + 1
      else                       ! Descend
        floor = floor - 1
      end if
      i = i + 1                  ! Get next character
    end do

    write(*,*) i - 1 ! Position starts at 1

  close(1)

end program notquitelisp